### PR TITLE
New package: plume-creator-0.66.1

### PR DESCRIPTION
**Pull request by VargMon**
I have testing it, and am also using it, on Void Linux x86_64 glibc... I had to make some minor modification to the hunspell template in order for it to compile and run correctly. This's package also contains the Debian+Ubuntu patches. The modification to the hunspell template is only provide libhunspell-1.6 as just 'libhunspell' which apparently the linker needed some help figuring out are the same lib.

Thank you,
VargMon (vargmon98 at gmail dot com)

### CODE REVIEW

**Comment by cr6git** [comment]
`maintainer="VargMon <VargMon@users.noreply.github.com>"`

No, maintainer needs a valid address.

File: srcpkgs/plume-creator/template [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+do_fetch() {
+    [ -f ${pkgname}_${_realversion}.orig.tar.xz ] || wget ${UBUNTU_SITE}/universe/p/plume-creator/${pkgname}_${_realversion}.orig.tar.xz
+    [ -f ${pkgname}_${_realversion}-${_debianbuild}.debian.tar.bz2 ] || wget ${UBUNTU_SITE}/universe/p/plume-creator/${pkgname}_${_realversion}-${_debianbuild}.debian.tar.bz2
+    echo f88f9524184bcf60dae855d6ad37899aefa39de0edd56098e93ce3b97b679e0c ${pkgname}_${_realversion}.orig.tar.xz > sha256sums
+    echo 7d473ee7283bc311c7d4dd3311d1e7726df7d45fa0123894265f355b436298e1 ${pkgname}_${_realversion}-${_debianbuild}.debian.tar.bz2>> sha256sums
+    sha256sum -c ./sha256sums
+    rm ./sha256sums
+    mkdir -p ${wrksrc}
+}
```
Comment (jnbr) [inline srcpkgs/plume-creator/template]: Use `distfiles` instead.

File: srcpkgs/plume-creator/template [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+do_extract() {
+    tar -C ../ -xvJf ../${pkgname}_${_realversion}.orig.tar.xz
+    tar -xvjf ../${pkgname}_${_realversion}-${_debianbuild}.debian.tar.bz2
+}
```
Comment (jnbr) [inline srcpkgs/plume-creator/template]: this is also covered by using `distfiles`

File: srcpkgs/plume-creator/template [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+pre_configure() {
+    cd ${wrksrc}
+}
+
+do_configure() {
+    # Bug for some reason the ld cache dosn't get updated and so we'll run ldconfig here
+    ldconfig -C ../../etc/ld.so.cache
+    sleep 1s
```
Comment (jnbr) [inline srcpkgs/plume-creator/template]: why?

**Comment by VargMon** [comment]
Sorry the address was added automatically by a script.... I just kind of forgot about the email... also that snip from the template file was to ensure that during the pre_configure stage the $PWD would be wrksrc.... for some unknown reason in my run of xbps-src, the build would hit an error unless I cd to wrksrc.... I’ll see if I can remove that... and I’ll update the email too... sorry about those :/

File: srcpkgs/plume-creator/template [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# Template file for 'plume-creator'
+pkgname=plume-creator
+version=0.66.1
+_realversion=0.66+dfsg1
+_debianbuild=3.1build2
+_gmake_cmd=make
+_qmake_cmd=qmake-qt5
+revision=1
+build_style=qmake
+wrksrc=${pkgname}-${version}
```
Comment (Johnnynator) [inline srcpkgs/plume-creator/template]: this is the default value of wrksrc.

File: srcpkgs/plume-creator/template [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+shlib_provides="libplume-creator-data.so.1 libplume-creator-gui.so.1 libplume-creator-qml.so.1"
```
Comment (Johnnynator) [inline srcpkgs/plume-creator/template]: does get automatically detected

File: srcpkgs/plume-creator/template [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    ldconfig -C ../../etc/ld.so.cache
+    sleep 1s
+    ${_qmake_cmd} plume-creator-all.pro
+}
+
+do_build() {
+    ln -s -T /usr/include/quazip ${wrksrc}/externals/quazip
+    # Bug for some reason the ld cache dosn't get updated and so we'll run ldconfig here again
+    ldconfig -C ../../etc/ld.so.cache
+    sleep 1s
```
Comment (Johnnynator) [inline srcpkgs/plume-creator/template]: do this in pre_build() and drop the do_build function

File: srcpkgs/plume-creator/template (hunk 3 above)
Comment (Johnnynator) [inline srcpkgs/plume-creator/template]: Move this to pre_configure

File: srcpkgs/plume-creator/template [hunk 7]
```diff
@@ -0,0 +1,63 @@
+# Template file for 'plume-creator'
+pkgname=plume-creator
+version=0.66.1
+_realversion=0.66+dfsg1
+_debianbuild=3.1build2
+_gmake_cmd=make
+_qmake_cmd=qmake-qt5
```
Comment (Johnnynator) [inline srcpkgs/plume-creator/template]: _qmake and _game shouldn't be needed.

File: srcpkgs/plume-creator/template (hunk 4 above)
Comment (VargMon) [reply srcpkgs/plume-creator/template]: True that, I'll fix it... I guess I was still trying  to write a Arch makepkg..

File: srcpkgs/plume-creator/template (hunk 5 above)
Comment (VargMon) [reply srcpkgs/plume-creator/template]: Glad you reminded me! This's actually leftovers from when I tried to build the git develop branch of Plume Creator.

File: srcpkgs/plume-creator/template (hunk 3 above)
Comment (Johnnynator) [reply srcpkgs/plume-creator/template]: you may have to add `configure_args="plume-creator-all.pro"`

File: srcpkgs/hunspell/template [hunk 8]
```diff
@@ -20,7 +22,10 @@ pre_configure() {
 libhunspell_package() {
 	short_desc+=" - runtime library"
 	pkg_install() {
-		vmove usr/lib/*.so.*
+        ln -s  -T ${pkgname}-${_sonamehyphenversion}.so.${_sonameversion} ${DESTDIR}/usr/lib/${pkgname}.so
+        ln -s  -T ${pkgname}-${_sonamehyphenversion}.so.${_sonameversion} ${DESTDIR}/usr/lib/${pkgname}.so.0
+		ln -s  -T ${pkgname}-${_sonamehyphenversion}.so.${_sonameversion} ${DESTDIR}/usr/lib/${pkgname}.so.${_sonameversion}
+		vmove usr/lib/*.so*
 	}
```
Comment (Johnnynator) [inline srcpkgs/hunspell/template]: the debian build-in-Qt5.patch adds the `-lhunspell` LDFLAG which should be `-lhunspell-1.6`. There is no need for this change in hunspell. Pume's buld system should just be patched acodingly

File: srcpkgs/plume-creator/template (hunk 3 above)
Comment (VargMon) [reply srcpkgs/plume-creator/template]: Ah thanks Johnnynator!

**Comment by VargMon** [comment]
Thanks all. Special thanks @Johnnynator. I've made the suggested modification.
I'll re-pull-request with the changes. :)

**Pull request closed without merging**